### PR TITLE
docs(policies): mesh proxy patch

### LINKED ISSUE #1196: MeshProxyPatch docs
### Description

A docs for this new policy

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
All examples are taken from the existing ProxyTemplate but adjusted for MeshProxyPatch

<Explain your change!>

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits)

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)?

Fix #1196

### CODE REVIEW

File: app/docs/dev/policies/meshproxypatch.md [hunk 1]
```diff
@@ -0,0 +1,862 @@
...
+              '@type': type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua
+              inline_code: |
+                function envoy_on_request(request_handle)
+                  request_handle:headers():add("x-header", "test")
+                end
+```
+{% endtab %}
+{% endtabs %}
+
+## Merging
```
Comment (lahabana) [inline app/docs/dev/policies/meshproxypatch.md:858]: Nit: Should this be before `example` ?

**Review by lahabana — APPROVED** [review]
LGTM just one point re: merging

**Comment by lahabana** [comment]
Also need to update: https://deploy-preview-1204--kuma.netlify.app/docs/dev/policies/introduction/ with the equivalence

**Comment by lahabana** [comment]
Please don't merge before https://github.com/kumahq/kuma-website/pull/1203 is in

File: app/docs/dev/policies/meshproxypatch.md [hunk 2]
```diff
@@ -0,0 +1,862 @@
...
+| `MeshService`       | ✅        |
+| `MeshServiceSubset` | ✅        |
+
+To learn more about the information in this table, see the [matching docs](/docs/{{ page.version }}/policies/targetref).
+
+## Configuration
+
+### Modifications
+
+`MeshProxyPatch` lets you specify modifications in `appendModification` that can add a new resource, patch an existing resource or remove an existing resource.
```
Comment (bartsmykla) [inline app/docs/dev/policies/meshproxypatch.md:36]: Maybe?
```suggestion
`MeshProxyPatch` lets you specify modifications in `appendModification` block that can add a new resource, patch an existing resource or remove an existing resource.
```

File: app/docs/dev/policies/meshproxypatch.md [hunk 3]
```diff
@@ -0,0 +1,862 @@
...
+
+Examples: add new filters but only on inbound listeners, set timeouts on outbound clusters.
+
+Well known origins:
+* `inbound` - resources generated for incoming traffic.
+* `outbound` - resources generated for outgoing traffic.
+* `transparent` - resources generated for transparent proxy functionality.
+* `prometheus` - resources generated for Prometheus to scrape when metrics on the Mesh is enabled.
+* `direct-access` - resources generated for Direct Access functionality.
+* `gateway` - resources generated for MeshGateway
```
Comment (bartsmykla) [inline app/docs/dev/policies/meshproxypatch.md:55]:
```suggestion
* `gateway` - resources generated for MeshGateway.
```

File: app/docs/dev/policies/meshproxypatch.md [hunk 4]
```diff
@@ -0,0 +1,862 @@
...
+
+The list is not complete, because policy plugins can introduce new resources.
```
Comment (bartsmykla) [inline app/docs/dev/policies/meshproxypatch.md:57]:
```suggestion
The list is not complete, as policy plugins can introduce new resources.
```

File: app/docs/dev/policies/meshproxypatch.md [hunk 5]
```diff
@@ -0,0 +1,862 @@
...
+* `AddAfter` - add a new filter after other filter in HTTP Connection Manager that is matched using `match` section.
+* `AddBefore` - add a new filter before other filter in HTTP Connection Manager that is matched using `match` section.
+* `Patch` - patch a matched filter in HTTP Connection Manager.
+* `Remove` - remove a filter in HTTP Connection Manager.
+
+Available matchers:
+* `name` - name of the network filter
+* `listenerName` - name of the listener
+* `listenerTags` - tags of inbound or outbound listeners. They match `Listener.metadata.filterMetadata[io.kuma.tags]` in XDS configuration.
+* `origin` - origin of the listener
```
Comment (bartsmykla) [inline app/docs/dev/policies/meshproxypatch.md:459]:
```suggestion
* `name` - name of the network filter.
* `listenerName` - name of the listener.
* `listenerTags` - tags of inbound or outbound listeners. They match `Listener.metadata.filterMetadata[io.kuma.tags]` in XDS configuration.
* `origin` - origin of the listener.
```
As other envoy related sections starts with capital letter, shouldn't we do the same here with `Listener`?

**Review by bartsmykla — APPROVED** [review]
few nits and questions

**Pull request merged**